### PR TITLE
Linting fixes

### PR DESCRIPTION
**Pull request by jonbarrow**
This PR fixes all linting errors reported by golangci-lint using our current configuration

Due to the nature of some of the linting issues, this also includes breaking changes and touches every source file and thus should be reviewed rather carefully

Linting errors fixed include

- Export stuttering
- Underscores in package names
- Underscores in variable names
- Ineffectual assignments
- Gosimple logic simplifications

Two of the biggest linting issues was export stuttering and underscores in package names

We historically named each protocol struct with it's protocol name because originally all code was in a single, root-level, package. Protocols were imported using `nexproto.SomethingNameProtocol`, and made sense for the time. As the library evolved we moved each protocol into it's own package, the package name being the protocol name

This package naming system made a lot of sense in context but broke several Go rules and is the root cause for these two large linter issues

In some cases protocol names are short nouns, as per Go's style guide. Such as `notifications`. However longer names, such as `match_making_ext`, and protocols with types and/or game specific patches, such as `datastore_super_smash_bros_4_types`, cannot be squashed down to short nouns without losing readability. These would become `matchmakingext` and `datastoresupersmashbros4types`

In other cases the protocol struct and the package names would stutter, such as `notifications.NotificationsProtocol`. Not only was this stuttering a linter issue, but the added `Notifications` part of the name is now redundant. When importing the package the developer has full context of what `Protocol` is, and this `notifications.Notifications` is no longer needed

The solution to both of these problems is the same: rename every package to have short noun package and struct names

Every protocol package, even game-specific patches, are now _all_ named `protocol`. If types exist, the package for that protocols types is simply named `types`

This has a few caveats

- The developer _must_ change the import name of each package to retain context. For example DataStore must now explicitly be imported as `datastore`. This is done pretty much everywhere already, however
- Our documentation site will no longer have detailed package names. Though the package names will be in context of the protocols folder
- When creating a new protocol that has game-specific patches, to retain context a new type alias must be made to embed the parent protocol into the child. Such as `type datastoreProtocol = datastore.Protocol`. `datastoreProtocol` would now be embedded into the struct and referenced via `protocol.datastoreProtocol`, rather than embedding `datastore.Protocol` directly and referencing via `protocol.Protocol` which loses context

### CODE REVIEW

File: ticket-granting/protocol.go [hunk 1]
```diff
@@ -31,8 +31,8 @@ const (
 	MethodLoginWithParam = 0x6
 )
 
-// TicketGrantingProtocol handles the Ticket Granting NEX protocol
-type TicketGrantingProtocol struct {
+// Protocol stores all the RMC method handlers for the Ticket Granting protocol and listens for requests
+type Protocol struct {
 	Server                *nex.Server
 	LoginHandler          func(err error, client *nex.Client, callID uint32, strUserName string)
```
Comment (DaniElectra) [inline ticket-granting/protocol.go:37]: Handlers should be unexported

**Review by DaniElectra — CHANGES_REQUESTED** [review]
Most of the review involves protocols that export handlers but don't need to be exported. These functions were already exported, but since this PR fixes linting, I think it fits here.

Aside from that and some minor comments for naming, looks good

**Comment by jonbarrow** [comment]
Sounds good. I was planning on unexporting these functions after this PR but you're right that it fits here too

**Comment by jonbarrow** [comment]
That should be all from that. Just waiting on the results from @SuperMarioDaBom. He said he would test these changes in a real server integration

**Review by SuperMarioDaBom — APPROVED** [review]
Did a test, and code compiled successfully! I don't believe these changes would affect the logic side of things, so it compiling is good enough of an indicator to me that it's working correctly.

**Pull request merged**